File: example-base-28/nodejs-utilities-playground-v2/src/dsa/sorting/bubble-sort.js
function bubbleSort(numbers) {
  const result = [...numbers];
  for (let i = 0; i < result.length - 1; i++) {
    let swapped = false;
    for (let j = 0; j < result.length - i - 1; j++) {
      if (result[j] > result[j + 1]) {
        [result[j], result[j + 1]] = [result[j + 1], result[j]];
        swapped = true;
      }
    }
    if (!swapped) break;
  }
  return result;
}
module.exports = { bubbleSort };

File: example-base-29/python-algorithms-playground-v2/algorithms/sorting/bubble_sort.py
def bubble_sort(arr):
    a = arr.copy()
    for i in range(len(a)):
        swapped = False
        for j in range(0, len(a) - i - 1):
            if a[j] > a[j + 1]:
                a[j], a[j + 1] = a[j + 1], a[j]
                swapped = True
        if not swapped:
            break
    return a

if __name__ == "__main__":
    print(bubble_sort([5, 1, 4, 2, 8]))
